drape/glyph.hpp

```diff
@@ -37,12 +31,40 @@ struct GlyphImage
   SharedBufferManager::shared_buffer_ptr_t m_data;
 };
 
+struct GlyphFontAndId
+{
+  int16_t m_fontIndex;
+  uint16_t m_glyphId;
+
+  // Required only for buffer_vector's internal T m_static[N];
+  GlyphFontAndId() = default;
+
+  constexpr GlyphFontAndId(int16_t fontIndex, uint16_t glyphId) : m_fontIndex(fontIndex), m_glyphId(glyphId) {}
+
+  bool operator==(GlyphFontAndId const & other) const
+  {
+    return m_fontIndex == other.m_fontIndex && m_glyphId == other.m_glyphId;
+  }
+
+  bool operator<(GlyphFontAndId const & other) const
+  {
+    return std::tie(m_fontIndex, m_glyphId) < std::tie (other.m_fontIndex, other.m_glyphId);
+  }
+};
+
+
+//using TGlyph = std::pair<int16_t /* fontIndex */, uint16_t /* glyphId */>;
+// TODO(AB): Measure if 32 is the best value here.
+using TGlyphs = buffer_vector<GlyphFontAndId, 32>;
```

Comment:
Note that GetGlyphs functions return by value -> so using move ctor -> the longer static buffer -> _stranger_ moving :)

- Use regular vector, especially with reserve
or
- Use GetGlyphs(TGlyphs & res) notation.

Reply:
There should be no move in the case of GetGlyphs. RVO optimizes any copying out.

Reply:
buffer_vector has a defined move ctor. Probably, you can rely on RVO in some very specific cases, but no in general.

Reply:
We are discussing a specific case, not "in general". The standard says that compilers are not obliged to apply NRVO, they may apply the optimization. This case fits the NRVO requirements. Are you aware of any of our compilers that do not use NRVO here?

With vector, we're guaranteed to have at least one allocation when using reserve (and more allocations without it). With NRVO there are no allocations.

Reply:
I think RVO won't be here. But as you wish, it's not a big deal IMO. I know how you like to "count matches", so here it is :)